packages/apputils-extension/src/shortcuts.tsx

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+
+import { JupyterFrontEnd } from '@jupyterlab/application';
+import { CommandRegistry } from '@lumino/commands';
+import { Selector } from '@lumino/domutils';
+import * as React from 'react';
+import { Dialog, showDialog } from '@jupyterlab/apputils';
+import { TranslationBundle } from '@jupyterlab/translation';
+/**
+ * The class name for each row of ContextShortcutTable
+ */
+const SHORTCUT_TABLE_ROW_CLASS = 'jp-ContextualShortcut-TableRow';
+/**
+ * The class name for the last row of ContextShortcutTable
+ */
+const SHORTCUT_TABLE_LAST_ROW_CLASS = 'jp-ContextualShortcut-TableLastRow';
+/**
+ * The class name for each item of ContextShortcutTable
+ */
+const SHORTCUT_TABLE_ITEM_CLASS = 'jp-ContextualShortcut-TableItem';
+/**
+ * The class name for each button-like symbol representing a key used in a shortcut in the ContextShortcutTable
+ */
+const SHORTCUT_KEY_CLASS = 'jp-ContextualShortcut-Key';
+
+export function displayShortcuts(
+  app: JupyterFrontEnd,
+  trans: TranslationBundle,
+  activeElement?: Element
+) {
+  const { commands } = app;
+  const elt = activeElement ?? document.activeElement;
+
+  /**
+   * Find the distance from the target node to the first matching node.
+   *
+   * Based on Lumino private function commands.Private.targetDistance
+   * This traverses the DOM path from `elt` to the root
+   * computes the distance from `elt` to the first node which matches
+   * the CSS selector. If no match is found, `-1` is returned.
+   *
+   * It also stops traversal if the `data-lm-suppress-shortcuts` or
+   * `data-p-suppress-shortcuts` attributes are found.
+   */
+
+  function formatKeys(keys: string[]): JSX.Element {
```

Comment:
Small note: it looks like this function is now separating the comment above from the function it applies to (the matchDistance function below). Can we move the comment down to be with its corresponding function again?

Reply:
@jasongrout Thanks for pointing this issue. This will be corrected in the next commit.